docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,941 @@
+:branch:          1.5
+:ls_version:		  1.5.0.beta1
+
+= How to write your own Logstash {plugintype} plugin
+
+This HOWTO will walk you through the process of developing a new {plugintype}
+for Logstash. The workflow has changed as of version 1.5 of Logstash, so you'll
+want to follow these new and improved steps to contribute a new plugin!
+
+
+{introfooter}
+
+Logstash is written in Ruby.  If you are unfamiliar with the Ruby programming
+language, an excellent quick-start guide is at
+https://www.ruby-lang.org/en/documentation/quickstart/.
+
+== Get started
+=== Create a new GitHub repo
+As of version 1.5 of Logstash, the code for each plugin lives in its own GitHub
+repository. So first, you'll need to create a new repo as a home for your
+nascent {plugintype}. When logged into GitHub, click on the "Repositories" tab.
+You may see a list of repositories you've forked and/or contributed to, or it
+may be blank. Either way, you're fine! Now click on the green "New" button near
+the top right of the page.
+
+Choose the following settings on your new repo:
+
+* Repository name should be similar to "logstash-{plugintype}-mypluginname"
+* Public or Private - up to you (but will need to be Public to submit as an
+  official plugin)
+* Click the box "Initialize this repository with a README"
+
+=== Copy the {plugintype} code
+
+==== Clone your forked repo
+You'll need to clone the new repo you just created. Use a command something like
+this:
+
+[source,sh]
+[subs="attributes"]
+----------------------------------
+git clone git@github.com:myuser/logstash-{plugintype}-mypluginname.git
+----------------------------------
+
+==== Clone and copy the example plugin (highly recommended)
+At this point, you could build the plugin out entirely on your own.  However,
```

Comment:
mention we follow standard rubygem dir structure....maybe show how a rubygem dir structure looks like:

from: http://timelessrepo.com/making-ruby-gems

Something like:
```
$ tree -d mygem
|-- lib/
|   |-- mygem.rb
|   `-- mygem/
|       |-- base.rb
|       `-- version.rb
|-- test/
|-- bin/
|-- Rakefile
`-- mygem.gemspec
```